src/ansys/geometry/core/tools/prepare_tools.py

```diff
@@ -0,0 +1,133 @@
+    get_bodies_from_ids,
+    get_design_from_body,
+)
+
+if TYPE_CHECKING:  # pragma: no cover
+    from ansys.geometry.core.designer.body import Body
+    from ansys.geometry.core.designer.edge import Edge
+    from ansys.geometry.core.designer.face import Face
+
+class PrepareTools:
+    """Prepare tools for PyAnsys Geometry."""
+
+    def __init__(self, grpc_client: GrpcClient):
+        """Initialize Prepare Tools class."""
+        self._grpc_client = grpc_client
+        self._prepare_stub = PrepareToolsStub(self._grpc_client.channel)
+
+    def extract_volume_from_faces(
+        self, sealing_faces: List["Face"],  inside_faces: List["Face"]
+    ) -> List["Body"]:
+        """Extract a volume from input faces.
+
+        Creates a volume (typically a flow volume) from a list of faces that seal the volume
+        and one or more faces that define the wetted surface (inside faces of the solid).
+
+        Parameters
+        ----------
+        sealing_faces : List[Face]
+            List of faces that seal the volume.
+        inside_faces : List[Face]
+            List of faces that define the interior of the solid.
+
+        Returns
+        -------
+        List[Body]
+            List of created bodies.
+        """
+        if not sealing_faces or not inside_faces:
+            return []
+
+        parent_body = sealing_faces[0].body
+        parent_design = get_design_from_body(parent_body)
+
+        sealing_faces_ids = [EntityIdentifier(id = face.id) for face in sealing_faces]
+        inside_faces_ids = [EntityIdentifier(id = face.id) for face in inside_faces]
+
+
+        volume_extract_response = self._prepare_stub.ExtractVolumeFromFaces
+        (ExtractVolumeFromFacesRequest(
+            sealing_faces=sealing_faces_ids,
+            inside_faces=inside_faces_ids)
+        )
+
+        bodies_ids = [created_body.id for created_body in volume_extract_response.created_bodies]
+        if (len (bodies_ids) > 0):
+            parent_design._update_design_inplace()
+        return get_bodies_from_ids(parent_design, bodies_ids)
+
+    def extract_volume_from_edge_loops(
+        self, sealing_edges: List["Edge"],  inside_faces: List["Face"]
+    ) -> List["Body"]:
+        """Extract a volume from input edge loops.
+
+        Creates a volume (typically a flow volume) from a list of edge loops that seal the volume.
+        and one or more faces that define the wetted surface (inside faces of the solid).
+
+        Parameters
+        ----------
+        sealing_edges : List[Edge]
+            List of faces that seal the volume.
+        inside_faces : List[Face]
+            List of faces that define the interior of the solid (Not always necessary).
+
+        Returns
+        -------
+        List[Body]
+            List of created bodies.
+        """
+        if not sealing_edges:
+            return []
+
+        parent_body = sealing_edges[0].faces[0].body
+        parent_design = get_design_from_body(parent_body)
```

Comment:
Also @smereu I just realized - there is also a method called ``get_design_from_edge`` and ``get_design_from_face`` in case you want to use them :) Much better than doing it manually yourself